presto-tests/src/test/java/com/facebook/presto/execution/TestStageCountWarningThreshold.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.QueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.execution.TestQueryRunnerUtil.assertWarnings;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.createQueryRunner;
+import static com.facebook.presto.spi.StandardWarningCode.TOO_MANY_STAGES;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestStageCountWarningThreshold
+{
+    private static final int STAGE_COUNT_WARNING_THRESHOLD = 20;
+    private QueryRunner queryRunner;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        queryRunner = createQueryRunner(ImmutableMap.of("query.stage-count-warning-threshold", String.valueOf(STAGE_COUNT_WARNING_THRESHOLD)));
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown()
+    {
+        queryRunner.close();
+        queryRunner = null;
+    }
+
+    @Test
+    public void testStageCountWarningThreshold()
+    {
+        StringBuilder queryBuilder = new StringBuilder("SELECT name FROM nation WHERE nationkey = 0");
+        String noWarningsQuery = queryBuilder.toString();
+        for (int key = 1; key <= STAGE_COUNT_WARNING_THRESHOLD; key++) {
```

Comment:
Is there a reason we're starting from `1` and using `<=`?

Reply:
To exceed the threshold by 1 since the queryBuilder is initialized with 1 stage.

Reply:
This is a fun example, so permit me to bikeshed a bit.

(1) `for (int key = 1; key <= STAGE_COUNT_WARNING_THRESHOLD; key++)`
(2) `for(int i = 0; i < STAGE_COUNT_WARNING_THRESHOLD; i++)`
(3) `for(int stageCount = 1; stageCount <= STAGE_COUNT_WARNING_THRESHOLD; stageCount++)`

(1) == (2) == (3) in terms of functionality
(2) > (1) because i = 0 is easier to read
(3) > (1) because stageCount is more descriptive than `key`

Reply:
Nice! Thanks for bikeshedding:) Seems like (3) is the best bike, will go with that one:)

Reply:
I modified the bike: stageCount is not accurate since it is off by 1. The variable is named stageIndex which reflects it's use more accurately.